src/bicep/mlz.bicep

```diff
@@ -1,30 +1,457 @@
+param operationsSubnetServiceEndpoints array = [
+  {
+    service: 'Microsoft.Storage'
+  }
+]
+
+// SHARED SERVICES PARAMETERS
+
+@description('An array of Network Diagnostic Logs to enable for the SharedServices Virtual Network. See https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=CMD#logs for valid settings.')
+param sharedServicesVirtualNetworkDiagnosticsLogs array = []
+
+@description('An array of Network Diagnostic Metrics to enable for the SharedServices Virtual Network. See https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=CMD#metrics for valid settings.')
+param sharedServicesVirtualNetworkDiagnosticsMetrics array = []
+
+@description('An array of Network Security Group rules to apply to the SharedServices Virtual Network. See https://docs.microsoft.com/en-us/azure/templates/microsoft.network/networksecuritygroups/securityrules?tabs=bicep#securityrulepropertiesformat for valid settings.')
+param sharedServicesNetworkSecurityGroupRules array = []
+
+@description('An array of Network Security Group diagnostic logs to apply to the SharedServices Virtual Network. See https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-nsg-manage-log#log-categories for valid settings.')
+param sharedServicesNetworkSecurityGroupDiagnosticsLogs array = [
+  {
+    category: 'NetworkSecurityGroupEvent'
+    enabled: true
+  }
+  {
+    category: 'NetworkSecurityGroupRuleCounter'
+    enabled: true
+  }
+]
+
+@description('An array of Network Security Group Diagnostic Metrics to enable for the SharedServices Virtual Network. See https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=CMD#metrics for valid settings.')
+param sharedServicesNetworkSecurityGroupDiagnosticsMetrics array = []
+
+@description('An array of Service Endpoints to enable for the SharedServices subnet. See https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-service-endpoints-overview for valid settings.')
+param sharedServicesSubnetServiceEndpoints array = [
+  {
+    service: 'Microsoft.Storage'
+  }
+]
+
+// LOGGING PARAMETERS
+
+@description('When set to "true", enables Microsoft Sentinel within the Log Analytics Workspace created in this deployment. It defaults to "false".')
+param deploySentinel bool = false
+
+@description('The daily quota for Log Analytics Workspace logs in Gigabytes. It defaults to "-1" for no quota.')
+param logAnalyticsWorkspaceCappingDailyQuotaGb int = -1
+
+@description('The number of days to retain Log Analytics Workspace logs. It defaults to "30".')
+param logAnalyticsWorkspaceRetentionInDays int = 30
+
+@description('The SKU for the Log Analytics Workspace. It defaults to "PerGB2018".')
+param logAnalyticsWorkspaceSkuName string = 'PerGB2018'
```

Comment:
If it's available, it would be good to provide a link to valid SKU names. There is an example here, but I don't see a definitive list. https://docs.microsoft.com/en-us/azure/azure-monitor/logs/resource-manager-workspace. This is not included in the scope of this PR so we could move it to another backlog item.

Reply:
Updated to use the recommended allowed values from the docs.microsoft.com link you provided. Technically, the most "correct" values are described in the API spec for the provider, but that's not very user friendly to digest and very specific to the current version of the provider we use in source: https://github.com/Azure/azure-rest-api-specs/blob/f398d20c96914e90fc2e77f115154abc8cb86346/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-06-01/Workspaces.json#L407-L416